lib/elixir/lib/range.ex

```diff
@@ -56,6 +56,41 @@ defmodule Range do
             "got: #{inspect(first)}..#{inspect(last)}"
   end
 
+  @doc """
+  Checks if two ranges overlap.
+
+  ## Examples
+
+      iex> Range.overlap?(1..7, 5..9)
+      true
+      iex> Range.overlap?(7..1, 5..9)
+      true
+      iex> Range.overlap?(1..7, 9..5)
+      true
+      iex> Range.overlap?(7..1, 9..5)
+      true
+      iex> Range.overlap?(0..1, 1..2)
+      true
+      iex> Range.overlap?(1..1, 1..1)
+      true
+      iex> Range.overlap?(1..5, 6..9)
+      false
+      iex> Range.overlap?(9..7, 3..1)
+      false
+  """
+  @spec overlap?(t, t) :: boolean
+  def overlap?(first1..last1, first2..last2) when first1 <= last1 and first2 <= last2 do
```

Comment:
Given `String.contains?`, should this be `overlaps?`? Or are we asking if "both ranges overlap"?

Reply:
Good question.

To me `overlap?` sounds more natural, similar to `MapSet.disjoint?` in a way, in which both arguments have the same importance, and the function is commutative. I think of them as "do the ranges overlap?", "are the sets disjoint?", "do the sets intersect?".

But I guess both options are equally good, you have a better perspective of naming conventions in the API, whatever you prefer 😄.

Reply:
Good points about `overlap?` being commutative.

However, I am still a bit worried about the `overlap?` not being plural and that being a possible inconsistency. Now that you added `disjoint?`, it is clear we are using it as a noun, that's why we can keep it singular. So I propose to either:

1. Figure out a name that is mostly seen as a noun
2. Keep only disjoint for now

Thoughts?

Reply:
I believe `overlap` is the idiomatic name for this, can't think of a good alternative. The variant that matches `disjoint?` would be `overlapping?` I believe ("these are disjoint intervals", "these are overlapping intervals"). But not sure I like how `Range.overlapping?(...)` reads.

Let's leave it at disjoint maybe?